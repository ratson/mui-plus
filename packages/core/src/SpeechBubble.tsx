import { Box, styled } from '@material-ui/core'
import { BoxProps } from '@material-ui/core/Box'

export type SpeechBubbleProps = BoxProps & {
  triangleLeft: string
}

const SpeechBubble = styled(Box)(({ theme, ...x }) => {
  const props = x as SpeechBubbleProps
  const backgroundColor = props.bgcolor || '#ccc'
  const borderRadius = props.borderRadius || '.4em'
  const triangleSize = '33px'
  const triangleLeft = props.triangleLeft || '50%'
  return {
    backgroundColor,
    position: 'relative',
    textAlign: 'center',
    padding: '20px 0',
    margin: '20px 30px',
    borderRadius,
    zIndex: 1,
    '&:after': {
      content: '""',
      position: 'absolute',
      display: 'block',
      bottom: 0,
      left: triangleLeft,
      marginLeft: `-${triangleSize}`,
      marginBottom: `-${triangleSize}`,
      width: 0,
      height: 0,
      border: `${triangleSize} solid transparent`,
      borderBottom: 0,
      borderTopColor: backgroundColor
    }
  }
}) as React.FC<SpeechBubbleProps>

export default SpeechBubble
