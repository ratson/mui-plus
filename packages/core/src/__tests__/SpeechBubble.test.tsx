import React from 'react'
import renderer from 'react-test-renderer'
import SpeechBubble from '../SpeechBubble'

describe('SpeechBubble', () => {
  test('render children', () => {
    const component = renderer.create(<SpeechBubble>Hello World</SpeechBubble>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
