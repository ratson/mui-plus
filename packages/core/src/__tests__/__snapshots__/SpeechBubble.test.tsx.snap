// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeechBubble render children 1`] = `
<div
  className="MuiBox-root MuiBox-root-4 Styled(MuiBox)-root-1 Styled(MuiBox)-root-2"
>
  Hello World
</div>
`;
